### ISSUE TITLE
Supporting GFDL and other models

### ISSUE THREAD

**Issue by jamesp** [xgcm/xgcm#34]
Hi all,

I'm working with the GFDL FMS idealised GCM and have been using `xarray` for a while now to perform most analysis of the model output.

The idea of a shared set of tools and functions for various models is great, not least because at some point I anticipate doing inter-model comparisons.

To that end, I was wondering if you would be interested in getting GFDL support into `xgcm`?  I have begun some (very preliminary!) work in a fork to make functions such as vertical integration model agnostic: see here for an idea of a base `GCMDataset` object (https://github.com/jamesp/xgcm/blob/master/xgcm/datasets/gcm_dataset.py).

I have a need to create functions for my work e.g. meridional mass streamfunction; I can either do this in a GFDL specific way, or with a bit more work, in an abstract GCM-agnostic way.

I guess I'm saying that if you'd like to support GFDL in `xgcm`, I'm happy to provide some time to do that. But if it's not part of your goals then I will continue to build with only GFDL in mind.

Cheers
James

**Comment by rabernat** [comment]
Sorry for the delayed reply.

There is an ongoing discussion about the concept of a general purpose GCM analysis package which began on the xarray mailing list:
https://groups.google.com/forum/#!topic/xarray/pv1d3txTLEw

and continues on this hackpad (which I just invited you to)
https://aospy.hackpad.com/

The oocgcm package from @lesommer is quickly eclipsing xgcm, and I imagine I will eventually abandon in 
https://github.com/lesommer/oocgcm

The bottom line is, we all agree that it would be silly to duplicate similar features in a dozen different packages and much better to collaborate on one package. But the best way to do that has not yet emerged. Your input and contributions would be welcome.

**Comment by jamesp** [comment]
Hi Ryan, thanks for the detailed response.

Seems like there are several independent streams with similar ideas.  I hadn't come across @lesommer's oocgcm package before, it looks promising.

Thanks for the invite to aospy hackpad, I'll close this issue and follow progress there.  Looking forward to contributing where I can.
